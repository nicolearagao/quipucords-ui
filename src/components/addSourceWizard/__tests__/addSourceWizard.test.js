import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { store } from '../../../redux';
import { AddSourceWizard } from '../addSourceWizard';

describe('AddSourceWizard Component', () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.spyOn(store, 'dispatch').mockImplementation((type, data) => ({ type, data }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a basic component', async () => {
    const props = {
      useGetAddSource: () => ({
        show: true
      })
    };

    const component = await shallowComponent(<AddSourceWizard {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should allow navigating to the next step on wizard fulfillment', () => {
    const mockStore = generateEmptyStore({
      addSourceWizard: {}
    });
    const props = {
      useGetAddSource: () => ({
        show: true,
        fulfilled: true
      })
    };

    const component = renderComponent(
      <Provider store={mockStore}>
        <AddSourceWizard {...props} />
      </Provider>
    );

    const input = component.screen.getByText('Next');
    component.fireEvent.click(input);
    expect(mockDispatch.mock.calls).toMatchSnapshot('next');
  });

  it('should allow closing the wizard on fulfillment', () => {
    const mockStore = generateEmptyStore({
      addSourceWizard: {}
    });
    const props = {
      useGetAddSource: () => ({
        show: true,
        fulfilled: true
      })
    };

    const component = renderComponent(
      <Provider store={mockStore}>
        <AddSourceWizard {...props} />
      </Provider>
    );

    const input = component.screen.getByLabelText('Close');
    component.fireEvent.click(input);
    expect(mockDispatch.mock.calls).toMatchSnapshot('close');
  });
});
