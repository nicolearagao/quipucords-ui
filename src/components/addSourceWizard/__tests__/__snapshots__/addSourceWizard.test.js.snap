// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSourceWizard Component should allow closing the wizard on fulfillment: close 1`] = `
[
  [
    {
      "source": {
        "source_type": "network",
      },
      "type": "VALID_SOURCE_WIZARD_STEPONE",
    },
  ],
  [
    [
      {
        "type": "CONFIRMATION_MODAL_HIDE",
      },
      {
        "type": "UPDATE_SOURCE_HIDE",
      },
    ],
  ],
  [
    {
      "type": "UPDATE_VIEW",
      "viewId": "sources",
    },
  ],
]
`;

exports[`AddSourceWizard Component should allow navigating to the next step on wizard fulfillment: next 1`] = `
[
  [
    {
      "source": {
        "source_type": "network",
      },
      "type": "VALID_SOURCE_WIZARD_STEPONE",
    },
  ],
]
`;

exports[`AddSourceWizard Component should render a basic component: basic 1`] = `
<Wizard
  className="quipucords-wizard__add-source"
  isForm={true}
  isNavHidden={false}
  isOpen={true}
  onClose={[Function]}
  onNext={[Function]}
  steps={
    [
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepOne)) />,
        "enableNext": false,
        "id": 1,
        "name": "t(form-dialog.title_add-source, {"context":"step"})",
      },
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepTwo)) />,
        "enableNext": false,
        "id": 2,
        "name": "t(form-dialog.title_add-source_step, {"context":"two"})",
        "nextButtonText": "t(form-dialog.label_submit, {"context":"add-source"})",
      },
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepThree)) />,
        "enableNext": false,
        "hideBackButton": false,
        "hideCancelButton": false,
        "id": 3,
        "name": "t(form-dialog.title_add-source_step, {"context":"three"})",
        "nextButtonText": "t(form-dialog.label, {"context":"close"})",
      },
    ]
  }
  title="t(form-dialog.title, {"context":"add-source"})"
  variant="medium"
/>
`;
